src/main/scala/sparkz/core/network/DeliveryTracker.scala

```diff
@@ -166,12 +176,12 @@ class DeliveryTracker(system: ActorSystem,
   def setReceived(id: ModifierId, sender: ConnectedPeer): Unit =
     tryWithLogging {
       val oldStatus: ModifiersStatus = status(id)
-      require(isCorrectTransition(oldStatus, Invalid), s"Illegal status transition: $oldStatus -> Received")
+      require(isCorrectTransition(oldStatus, Received), s"Illegal status transition: $oldStatus -> Received")
       if (oldStatus != Received) {
         requested(id).cancellable.cancel()
         requested.remove(id)
         decrementPeerLimitCounter(sender)
-        received.put(id, sender)
+        received.put(id, (sender, System.nanoTime()))
```

Comment:
NTP is using `System.currentTimeMillis()` which is not recommended for measuring execution time. As per the description:

System.currentTimeMillis() will give you the most accurate possible elapsed time in milliseconds since the epoch, but System.nanoTime() gives you a nanosecond-precise time, relative to some arbitrary point.

I think we won't have any benefit from NTP here, and also I think either won't be a problem for us because of the magnitude of TimeUnits we are interested in.